lib/screens/scan_and_convert.dart

```diff
@@ -1,63 +1,35 @@
-import 'dart:io';
+import 'package:get/get.dart';
+import 'package:flutter/material.dart';
 
+import 'package:EasyScan/controllers/scan_and_convert.dart';
 import 'package:EasyScan/Utils/constants.dart';
-import 'package:EasyScan/Utils/methods.dart';
-
-import 'package:flutter/material.dart';
 
-import 'package:image_picker/image_picker.dart';
+class ScanAndConvert extends StatelessWidget {
+  final scanAndConvertController = Get.put(ScanAndConvertController());
 
-class ScanAndConvert extends StatefulWidget {
-  @override
-  _ScanAndConvertState createState() => _ScanAndConvertState();
-}
-
-class _ScanAndConvertState extends State<ScanAndConvert> {
-  bool _hasNotPickedImage = false;
   Widget get getBody {
-    if (_hasNotPickedImage) {
+    if (scanAndConvertController.hasNotPickedImage) {
       return Center(
         child: RaisedButton(
           textColor: Colors.white,
-          onPressed: () => _getImageFromSource(),
+          onPressed: () => scanAndConvertController.getImageFromSource(),
           child: const Text('Open Camera'),
         ),
       );
     }
-    if (_imageFile == null) {
+    if (scanAndConvertController.imageFile == null) {
       return const Center(
           child: CircularProgressIndicator(
         valueColor: AlwaysStoppedAnimation<Color>(primaryColor),
       ));
     } else {
-      return Image.file(_imageFile);
-    }
-  }
-
-  File _imageFile;
-
-  @override
-  void initState() {
-    super.initState();
-    _getImageFromSource();
-  }
-
-  Future<void> _getImageFromSource() async {
-    final PickedFile _pickedFile =
-        await ImagePicker().getImage(source: ImageSource.camera);
-    if (_pickedFile != null) {
-      cropImage(_pickedFile.path, (path) {
-        _imageFile = File(path);
-        //TODO:send to editing page (first make one)
-      });
-    } else {
-      _hasNotPickedImage = true;
+      return Image.file(scanAndConvertController.imageFile);
     }
-    setState(() {});
   }
 
   @override
   Widget build(BuildContext context) {
+    scanAndConvertController.getImageFromSource();
```

Comment:
since we don't have tests in this project.So can you just verify this don't cause stackoverflow

Reply:
I have verified it, only the body listens to changes and build method runs just once. Though, I tweaked it in latest commit to remove confusions.